### PR TITLE
[18.0][MIG] mail_tracking: Migration to 18.0

### PR DESCRIPTION
**Pull request by trisdoan**
### Note
- missing commits from 14.0 are ported in https://github.com/trisdoan/social/pull/20

#### Backend
- `access_rights_uid` was removed from `_search()` changed in https://github.com/odoo/odoo/pull/171374/commits/21d023e6bc00793ac3b016cfdcafd05c9a81d95d
- `_prepare_outgoing_list` got extra param `mail_server` in https://github.com/odoo/odoo/pull/96008/commits/97ce0844ca0c3d531a7a2e2e47a867d89835b24e
- `_message_format` was converted to use store and `message_format_extras` was removed in https://github.com/odoo/odoo/commit/52df49eabb6e44ea6e0ed30b61e5a555af2e606a. As consequences, `_extras_to_store` is introduced
- `tools.ustr` is considered to be inefficient and flagged as deprecated in https://github.com/odoo/odoo/pull/173811/commits/64a5724e88de6532cf2a04e9243d16b079319ffd. Hence, is replaced by `exception_to_unicode` for the same effect.

#### Frontend
- `mail.messaging` service was removed in https://github.com/odoo/odoo/commit/683258b28107f44fba363e3b4e69f181cac3f80e
- `initMessagingCallback` is removed in https://github.com/odoo/odoo/commit/4f8087f03ad904b99f8f3580557a3ea744a97b76. So new entry point is created for store `onStarted` instead.
- mail.thread service was removed in  https://github.com/odoo/odoo/commit/65e5a7943de094fe538f2b978cb58c028a5694a1

### This changes
- Store helper was introduced in https://github.com/odoo/odoo/pull/150625/commits/c032ccfbaef91da0ee244f0e44e0e4e9d8d8d3fb. As consequences, failed data is now passed through Store.
   - As a side effect, it is also easier to do the search filter_failed(thread) with Store service
  
- As Odoo uses less Jquery, the module runs fine with vanilla. This impacts `mail_tracking/static/src/components/message_tracking/message_tracking.esm.js`

### Result
discuss_sidebar

failed_search_panel

failed_message_review

### CODE REVIEW

File: mail_tracking/static/src/components/failed_message/failed_message.esm.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
+/** @odoo-module **/
```
Comment (nilshamerlinck) [inline mail_tracking/static/src/components/failed_message/failed_message.esm.js]:
```suggestion
```

Comment (nilshamerlinck) [reply mail_tracking/static/src/components/failed_message/failed_message.esm.js]: not needed anymore

File: mail_tracking/models/mail_message.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        """Returns the list of failed messages to be used by the
+        failed_messages widget"""
+        return [
+            msg._prepare_dict_failed_message()
+            for msg in self.sorted("date", reverse=True)
+        ]
+
+    def set_need_action_done(self):
+        """This will mark the messages to be ignored in the tracking issues filter"""
+        self.check_access_rule("read")
```
Comment (nilshamerlinck) [inline mail_tracking/models/mail_message.py]: deprecated in favor of `check_access()`

File: mail_tracking/models/mail_tracking_email.py [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+                "(Document type: %(desc)s, Operation: %(operation)s)"
+            )
+            % {"desc": self._description, "operation": operation}
+            + " - ({} {}, {} {})".format(
+                _("Records:"), list(disallowed_ids), _("User:"), self._uid
+            )
+        )
+
+    def read(self, fields=None, load="_classic_read"):
+        """Override to explicitly call check_access_rule, that is not called
```
Comment (nilshamerlinck) [inline mail_tracking/models/mail_tracking_email.py]: is it still needed?

File: mail_tracking/models/mail_tracking_email.py [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+            values.update(
+                {
+                    "error_type": "no_recipient",
+                    "error_description": "The partner doesn't have a defined email",
+                }
+            )
+        else:
+            values.update(
+                {
+                    "error_smtp_server": tools.exception_to_unicode(smtp_server),
```
Comment (nilshamerlinck) [inline mail_tracking/models/mail_tracking_email.py]: is smtp_server an exception?

**Review by nilshamerlinck — CHANGES_REQUESTED** [review]

**Comment by trisdoan** [comment]
DRAFT: working on implementing permissions for mail.tracking.email

### Context
- This [commit](https://github.com/OCA/social/pull/966/commits/b4ab1600fe0b464937ff3dc06ff7cbf3a29afe3a)  porting from 12.0 implements permission constraint
- I am refactoring it with new API + adding tests
- Take this occasion to resolve the FIXME

**Comment by trisdoan** [comment]
superseded by https://github.com/trisdoan/social/pull/22

File: mail_tracking/models/mail_tracking_email.py (hunk 4 above)
Comment (trisdoan) [reply mail_tracking/models/mail_tracking_email.py]: yeah u'r right, how silly I am, it's just a name

wonder why the author uses it :sweat_smile:

**Comment by trisdoan** [comment]
Hi anh @nilshamerlinck, I added a patch to include your suggestion in the new PR

Please help to review :pray:

**Pull request closed without merging**